### PR TITLE
Reports on toggle protections off

### PR DESCRIPTION
**Pull request by jaceklyp**
**Required**:

Task/Issue URL: https://app.asana.com/0/72649045549333/1205734628204384/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2536
macOS PR: https://github.com/duckduckgo/macos-browser
What kind of version bump will this require?: Major

**Optional**:

Tech Design URL:
CC: @shakyShane 

**Description**:
Send a simplified breakage report through Pixel when user disables protections on iOS and macOS as such volume information from the toggle can help us identify breakage beyond what we can learn from regular breakage reports.

We aim to:
Explicitly request permission for such report without directing the user toward a specific response.
Be transparent by openly listing the exact information we intend to send.

**Steps to test this PR**:
See: https://app.asana.com/0/0/1206747477156794/f

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/BrowserServicesKit/ContentBlocking/ToggleProtectionsCounter.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        public static let onCountKey = "ToggleProtectionsCounter_On_Count"
+        public static let offCountKey = "ToggleProtectionsCounter_Off_Count"
+        public static let lastSendAtKey = "ToggleProtectionsCounter_Date"
+        public static let sendInterval: Double = 60 * 60 * 24 // 24 hours
+
+    }
+
+    private let store: KeyValueStoring
+    private let sendInterval: Double
```
Comment (ayoy) [inline Sources/BrowserServicesKit/ContentBlocking/ToggleProtectionsCounter.swift:49]: or `TimeInterval` 👨‍🍳 💋

File: Sources/BrowserServicesKit/ContentBlocking/ToggleProtectionsCounter.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        sendEventsIfNeeded(currentTime: currentTime)
+    }
+
+    public func sendEventsIfNeeded(currentTime: Date = Date()) {
+        guard let lastSendAt else {
+            save(lastSendAt: currentTime)
+            return
+        }
+
+        if abs(currentTime.timeIntervalSince(lastSendAt)) > sendInterval {
```
Comment (ayoy) [inline Sources/BrowserServicesKit/ContentBlocking/ToggleProtectionsCounter.swift:76]: Is `abs` really needed here?

File: Sources/Common/Extensions/BoolExtension.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+import Foundation
+
+public extension Bool {
+
+    var stringValue: String { self ? "true" : "false" }
```
Comment (ayoy) [inline Sources/Common/Extensions/BoolExtension.swift:22]: `self.description` provides the same functionality actually
```
$ swift repl
Welcome to Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5).
Type :help for assistance.
  1> true.description
$R0: String = "true"
  2> false.description
$R1: String = "false"
```

File: Sources/PrivacyDashboard/ToggleReportsFeature.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+public struct ToggleReportsFeature {
+
+    public var privacyConfiguration: PrivacyConfiguration
+    public init(privacyConfiguration: PrivacyConfiguration) {
+        self.privacyConfiguration = privacyConfiguration
+    }
+
+    public var isEnabled: Bool {
+        return true
```
Comment (jaceklyp) [inline Sources/PrivacyDashboard/ToggleReportsFeature.swift:30]: it is temporary

Comment (ayoy) [reply Sources/PrivacyDashboard/ToggleReportsFeature.swift:30]: I hope 😂

**Review by ayoy — APPROVED** [review]
LGTM! Leaving a few really minor comments, otherwise it looks solid!

**Review by ayoy — APPROVED** [review]
LGTM 👍

**Pull request merged**